src/Disks/IO/AsynchronousBoundedReadBuffer.cpp

```diff
@@ -231,13 +233,9 @@ bool AsynchronousBoundedReadBuffer::nextImpl()
         pos = working_buffer.begin();
     }
 
-    file_offset_of_buffer_end = impl->getFileOffsetOfBufferEnd();
+    file_offset_of_buffer_end += bytes_to_ignore + available();
     bytes_to_ignore = 0;
-
-    /// In case of multiple files for the same file in clickhouse (i.e. log family)
-    /// file_offset_of_buffer_end will not match getImplementationBufferOffset()
-    /// so we use [impl->getImplementationBufferOffset(), impl->getFileSize()]
-    chassert(file_offset_of_buffer_end <= impl->getFileSize());
```

Comment:
why remove this assertion? seems a valid assertion to me, plus I have seen it fail and it have caught a problem